pkg/pdp/service/roots_add.go

```diff
@@ -383,105 +367,27 @@ func (p *PDPService) AddRoots(ctx context.Context, id uint64, request []types.Ro
 			return common.Hash{}, fmt.Errorf("computing height and padding: %w", err)
 		}
 		// NB: defined here: https://github.com/FilOzone/pdp/blob/main/src/PDPVerifier.sol#L44
-		// as MAX_PIECE_SIZE_LOG2 = 50
-		// TODO: there is an accessor method on the verifier for getting this value, add to interface
-		if height > 50 {
+		maxPieceSizeLog2, err := p.verifierContract.MaxPieceSizeLog2(ctx)
+		if err != nil {
+			return common.Hash{}, err
+		}
+		if uint64(height) > maxPieceSizeLog2.Uint64() {
 			return common.Hash{}, fmt.Errorf("invalid height: %d", height)
 		}
 
-		// REVIEW: I am mega unsure of these code changes, but they do "pass" with the PDP Verifier contract.
-		// TODO(forrest): since migrating to the new contract, rod states there is an error
-		// in the curio size logic, context here: https://github.com/filecoin-project/curio/issues/650
-		// in commit 91aff56959407ec83171ef73d48c51fed8afb4c7 of curio
-		// Filecoin Team hash stated:
-		// subpieces are broken on our branch (called `rename` in Curio, don’t use it if you’re relying on subpieces.
-		// In fact, PieceCIDv2 isn’t quite working on our branch with this being an outstanding problem that’s currently
-		// biting us: https://github.com/filecoin-project/curio/issues/650; the not-quite-comprehensive switch to
-		// PieceCIDv2 hasn’t been a success, but we don’t really want to be changing the db schema just for that so we
-		// compromised and deferred the proper job to mkv2.
-
-		// Get total size by summing up the sizes of subroots
-		// We track both padded and unpadded sizes:
-		// - Padded sizes are used for smart contract operations
-		// - Unpadded sizes are used to validate against PieceCIDV2's embedded rawSize
 		var totalSize uint64 = 0
-		var totalUnpaddedSize uint64 = 0
 		var prevSubrootSize = subrootInfoMap[addRootReq.SubRoots[0]].PieceInfo.Size
 		for i, subrootEntry := range addRootReq.SubRoots {
 			subrootInfo := subrootInfoMap[subrootEntry]
 			if subrootInfo.PieceInfo.Size > prevSubrootSize {
+				// implies a bad request
 				return common.Hash{}, fmt.Errorf("subroots must be in descending order of size, root %d %s is larger than prev subroot %s", i, subrootEntry, addRootReq.SubRoots[i-1])
 			}
 
-			prevSubrootSize = subrootInfo.PieceInfo.Size
```

Comment:
We should update this no? Otherwise we're always comparing against the first piece not the previous piece.

Reply:
ahh great catch, yeah this is important for validation, will fix.